website/blog/2024-09-30-detox-copilot-is-out.md

```diff
@@ -0,0 +1,219 @@
+
+### Seamless Integration with Detox
+
+Detox Copilot is built into Detox and requires no additional installation. Simply initialize it in your test setup, and you're ready to start writing natural language tests.
+
+### LLM-Agnostic Design
+
+While Detox Copilot uses LLMs to interpret instructions, it is designed to be LLM-agnostic. This means you can connect it to your preferred language model service, offering flexibility and future-proofing your testing strategy.
+
+## Getting Started with Detox Copilot
+
+### Step 1: Initialize Detox Copilot
+
+Initialize Detox Copilot in your test setup by implementing a `PromptHandler` that connects to your LLM service. This abstraction allows Detox Copilot to communicate with any LLM you choose, promoting versatility across different testing frameworks.
+
+In our [Testing with Copilot guide](https://wix.github.io/Detox/docs/next/guide/testing-with-copilot), you can find implementation examples and more detailed descriptions for this part.
+
+### Step 2: Write Tests Using Natural Language
+
+Start writing your tests using the `copilot.perform` method with natural language steps.
+
+**Example:**
+
+```javascript
+it('should log in successfully', async () => {
+  await copilot.perform(
+    'Launch the app',
+    'Tap on the "Login" button',
+    'Enter "user@example.com" into the "Email" field',
+    'Enter "securePassword" into the "Password" field',
+    'Tap on the "Submit" button',
+    'Verify that the "Welcome Back!" message is displayed'
+  );
+});
+```
+
+You can also break Copilot commands into parts and use the traditional Detox APIs in a hybrid way wherever needed.
+This allows you to leverage both the simplicity of natural language commands and the control of standard Detox methods.
+
+**Hybrid Example:**
+
+```javascript
+it('should add an item to the cart', async () => {
+  // Use Copilot for initial steps
+  await copilot.perform(
+    'Launch the app',
+    'Navigate to the "Products" page',
+    'Tap on the "Add to Cart" button for the first product'
+  );
+
+  // Use traditional Detox APIs for hard-coded assertions
+  const cartBadge = element(by.id('cart-badge'));
+  await expect(cartBadge).toHaveText('1');
+  
+  // Continue with Copilot
+  await copilot.perform(
+    'Navigate to the "Cart" page',
+    'Verify that the product is listed in the cart'
+  );
+});
+```
+
+You can also ask Copilot to return values at the end of its execution and use them in your hard-coded assertions.
+For example, you might want to retrieve the title text of a page and perform assertions on it.
+
+**Returning Values Example:**
+
+```javascript
+it('should display the correct page title', async () => {
+  // Use Copilot to navigate and retrieve the page title
+  const pageTitle = await copilot.perform(
+    'Launch the app',
+    'Navigate to the "Profile" page',
+    'Return the text of the page title'
+  );
+
+  // Use the returned value in a hard-coded assertion
+  jestExpect(pageTitle).toBe('Profile');
```

Comment:
Hmm isn't this discouraged? I wonder

Reply:
hmm, yeahhh.. you're right. not the best example for this ability. I'll try to come with a better one